### PR TITLE
ZippedToolkitContext now works for build farm submission.

### PR DESCRIPTION
**Pull request by wmarshall484**
The ZippedToolkitContext will create a zip file containing the following:
  * The topology toolkit
  * The generated toolkit with user code
  * A makefile which will build the .sab file when invoked
  * A manifest of the toolkits in the archive (currently just two mentioned)
  * A "main compostite" file, containing the name of the main composite, and its namespace.

### CODE REVIEW

File: com.ibm.streamsx.topology/opt/python/templates/common/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+all:
+	chmod 777 com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/pyversion.sh
```
Comment (ddebrunner) [inline com.ibm.streamsx.topology/opt/python/templates/common/Makefile]: Only change permissions for the user, e.g. `500`  or `u+x`

**Comment by wmarshall484** [comment]
test now

File: java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java [hunk 2]
```diff
@@ -51,31 +68,75 @@ public Type getType() {
...
+		List<Path> paths = new ArrayList<Path>();
+		paths.add(Paths.get(topologyToolkit.getAbsolutePath()));
+		paths.add(Paths.get(workingDir + "/manifest_tk.txt"));
+		paths.add(Paths.get(workingDir + "/main_composite.txt"));
+		paths.add(Paths.get(topologyToolkit.getAbsolutePath() + "/opt/python/templates/common/Makefile"));
+		paths.add(folder);
+		
+		addAllToZippedArchive(paths, zipFilePath);	
+		JSONObject jso = new OrderedJSONObject();
+		super.deleteToolkit(new File(paths.get(1).toString()), jso);
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java]: Why not just `new File(paths.get(1).toString()).delete()` - no need to use JSON then.

Comment (wmarshall484) [inline java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java]: Fixed. At one point, I was deleting a directory, which required the *fileWalkTree* delete offered by `deleteToolkit`.

File: java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java [hunk 3]
```diff
@@ -99,9 +99,8 @@ public Path pack(final Path folder, String namespace, String name, String tkName
 		paths.add(folder);
 		
 		addAllToZippedArchive(paths, zipFilePath);	
-		JSONObject jso = new OrderedJSONObject();
-		super.deleteToolkit(new File(paths.get(1).toString()), jso);
-		super.deleteToolkit(new File(paths.get(2).toString()), jso);
+		new File(paths.get(1).toString()).delete();
+		new File(paths.get(2).toString()).delete();
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java]: Minor point but just realised it's much simpler:

`paths.get(1).toFile().delete()`

Comment (wmarshall484) [inline java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java]: Fixed.

**Pull request merged**